alacritty_terminal/src/tty/unix.rs

```diff
@@ -184,6 +185,16 @@ fn default_shell_command(shell: &str, user: &str) -> Command {
 pub fn new(config: &Options, window_size: WindowSize, window_id: u64) -> Result<Pty> {
     let pty = openpty(None, Some(&window_size.to_winsize()))?;
     let (master, slave) = (pty.controller, pty.user);
+    from_file_descriptors(config, window_id, master, slave)
+}
+
+/// Create a new TTY from a Pty's file descriptors.
+pub fn from_file_descriptors(
```

Comment:
I think `fd` is actually more universally used than `file_descriptor`.

```suggestion
pub fn from_fd(
```